packages/firestore/src/remote/remote_store.ts

```diff
@@ -228,15 +228,26 @@ export class RemoteStore {
         this.startWatchStream();
       }
 
-      this.updateAndBroadcastOnlineState(OnlineState.Unknown);
+      this.setOnlineStateToUnknown();
 
       return this.fillWritePipeline(); // This may start the writeStream.
     });
   }
 
-  /** Temporarily disables the network. The network can be re-enabled using enableNetwork(). */
-  disableNetwork(): Promise<void> {
-    this.updateAndBroadcastOnlineState(OnlineState.Failed);
+  /**
+   * Temporarily disables the network. The network can be re-enabled using
+   * enableNetwork(). By default this will also update (and broadcast) our
+   * onlineState to OnlineState.Failed. If this should not happen (e.g. because
+   * this is being called as part of shutdown logic or because we're going to
+   * immediately re-enable the network), pass `true` for
+   * suppressOnlineStateFailure.
+   */
+  disableNetwork(suppressOnlineStateFailure?: boolean): Promise<void> {
```

Comment:
`suppressOnlineStateNotification` ?

Reply:
We only suppress failure (if you pass false, we still trigger the Unknown state).